tests/pytest/test_cocotb.py

```diff
@@ -4,8 +4,12 @@
 
 import os
 
+import pytest
+
 from cocotb.runner import get_runner
 
+pytestmark = pytest.mark.simulator_required
```

Comment:
A simpler solution is just just move this into a different directory.

Reply:
I prefer solutions that don't require looking at documentation. When it comes to complexity "ounces make pounds". If it brings actual value over the other possible solutions then we can leave it.

Reply:
> I prefer solutions that don't require looking at documentation.

Well, either way you have to know something: either you have to know that tests which require a simulator need to go into a specific directory, or you need to know that a marker is needed.

I initially had a comment on top of the line, but removed it, because it read "This test requires a simulator", which is exactly what the line below says. I'll create a wiki page (after merging this change) to document some useful test writing tips.